meddlr/data/transforms/2D_motion.py

```diff
@@ -0,0 +1,87 @@
+from typing import Optional, Tuple, Sequence, Union
+
+import math
+import numpy as np
+import torch
+import meddlr.ops as F
+from meddlr.transforms import RandomAffine, RandomTranslation, TransformList
+from meddlr.forward.mri import SenseModel
+
+from meddlr.utils.events import get_event_storage
+
+
+class MotionModel2D:
+    """A model that corrupts kspace inputs with motion.
+
+    Motion is a common artifact experienced during the MR imaging forward problem.
+    When a patient moves, the recorded (expected) location of the kspace sample is
+    different than the actual location where the kspace sample that was acquired.
+    This module is responsible for simulating different motion artifacts.
+    """
+
+    def __init__(self, nshots, angle, translate, trajectory, seed=None):
+        """
+        Args:
+            nshots (int) : The number of shots in the image. 
+                This should be equivalent to ceil(phase_encode_dim / 
+                echo_train_length). 
+            angle : The (min, max) angle for rotation. Values should be in 
+                degrees and should be >=-180, <=180. Use 'None' to 
+                ignore rotation.
+            translate: The fraction of (height, width) to translate. 
+                e.g. 0.1 => 10% of the corresponding dimension.
+                So (0.1, 0.2) => 10% of height, 20% of width.
+                Use 'None' to ignore translation. 
+            trajectory: One of 'interleaved' or 'consecutive'. 
+        """
+        self.nshots = nshots
+        self.angle = angle 
+        self.translate = translate 
+        self.trajectory = trajectory
+        self.seed = seed
+
+    def __call__(self, *args, **kwargs):
+        return self.forward(*args, **kwargs)
+
+    def forward(self, image) -> torch.Tensor:
+        """
+        Simulate 2D motion for multi-shot Cartesian MRI.
+
+        This function supports two trajectories:
+        - 'blocked' : Where each shot corresponds to a consecutive block of 
+          kspace. (e.g. 1 1 2 2 3 3)
+        - 'interleaved' : Where shots are interleaved (e.g. 1 2 3 1 2 3) 
+        
+        We assume the phase encode direction is left to right (i.e. along
+        width dimesion). 
+
+        TODO: Add support for sensitivity maps. 
```

Comment:
these comments don't apply for this function - move them to the class docstring level